src/tests/nativeaot/SmokeTests/HelloWasm/HelloWasm.cs

```diff
@@ -40,7 +40,9 @@ private static unsafe int Main(string[] args)
         TestMetaData();
 
         TestGC();
+#if !NO_EXPLICIT_FINALIZER
         TestFinalization();
+#endif
```

Comment:
It's best to fix the finalizer in some following change so that the test doesn't hang instead (it'll need to be done eventually anyway) (or it may end up being the case that the test itseld needs fixing).
```suggestion
        TestFinalization();
```